### PR TITLE
Resisting out of cocoon

### PR DESCRIPTION
**Pull request by adacovsk**
If you manage to survive being wrapped in a cocoon, there is no way to escape out of it. This mainly affects zombies, vampires, and changelings, being permanently confined in one, with no hope of ever escaping. Not very fun

## What this does
- You can resist out of a cocoon. It takes two minutes
- A flat 80 tox damage... No more clutch ZAS cocooning instantly killing you. So the spider has to hit you at least once or twice before you go into crit and die

## Why it's good
- No instant death from cocooning
- No cuck prison for eternity from cocooning

## Changelog

:cl:
 * tweak: Spider cocoons have gotten weaker. You can resist out of them if you survive!

### CODE REVIEW

**Comment by jwhitak** [comment]
~~Please rename the PR. "Resisting out of cocoon" PR changing spider damage from clone to tox is the literal definition of sneaky coding.~~

My bad. I misunderstood what the spider damage change did. It still fits the title to the PR since it only affects spider attacks on already cocooned targets.

**Comment by adacovsk** [comment]
> ~Please rename the PR. "Resisting out of cocoon" PR changing spider damage from clone to tox is the literal definition of sneaky coding.~
> 
> My bad. I misunderstood what the spider damage change did. It still fits the title to the PR since it only affects spider attacks on already cocooned targets.

you're still wrong, it's just the cocoon damage

**Comment by BomberBro** [comment]
can't you just attack the cocoon to roll out of it on the extremely unlikely scenario you revive inside (so basically just changeling/vampire)

**Comment by adacovsk** [comment]
> can't you just attack the cocoon to roll out of it on the extremely unlikely scenario you revive inside (so basically just changeling/vampire)

That requires a second person. You can't bite (zombie) through it either

**Comment by BomberBro** [comment]
no I've definitely been rejuv'd inside a cocoon and punched my way out by clicking on it, has that somehow changed in the last couple months?

**Comment by WindowsErrors** [comment]
regardless if you can or can't claw your way out of the cocoon there should be some alternative command like this

**Comment by Killette2** [comment]
If Bomberbro claims he was able to punch out of it this should be checked for. It might be one of those things that you can do but the game provides no feedback. Other than that I see no problem with it taking 2 minutes to rip out of a spider cocoon if you're somehow still alive, but a glass shard in the pocket hitting the cocoon til it breaks alternative would also be interactive if we had it alongside

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -838,6 +838,14 @@ Thanks.
 			qdel(package)
 			playsound(src.loc, 'sound/items/poster_ripped.ogg', 100, 1)
 		return
+	else if(istype(src.loc, /obj/effect/spider/cocoon))
+		var/obj/effect/spider/cocoon/cocoon = src.loc
+		to_chat(L, "<span class='warning'>You attempt to untangle yourself, the webs are tight and will take some time.</span>")
+		if(do_after(src, src, 2 MINUTES))
```
Comment (Kurfursten) [inline code/modules/mob/living/living.dm:844]: user should still be L

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -838,6 +838,14 @@ Thanks.
...
+			L.visible_message("<span class='danger'>[L] successfully breaks out of [cocoon]!</span>",\
+							  "<span class='notice'>You successfully break out!</span>")
+			forceMove(T)
```
Comment (Kurfursten) [inline code/modules/mob/living/living.dm:847]: shouldn't this be L.forceMove?

Comment (adacovsk) [reply code/modules/mob/living/living.dm:847]: I'm just copying the other code that's already there and it works

Comment (Kurfursten) [reply code/modules/mob/living/living.dm:847]: What are you trying to do with this line of code? It appears to move the cocoon to the tile that the cocoon is on.

Comment (adacovsk) [reply code/modules/mob/living/living.dm:847]: I couldn't tell you what the code does, I just know it works and I copied it. I really can't say more about it, sorry

Comment (ShiftyRail) [reply code/modules/mob/living/living.dm:847]: Well here `src` is `usr` and is also the guy doing resist, so all that using `L` does is a typecheck.
It's probably fine. See L794.

**Pull request merged**